### DISCUSSION TITLE
Automatically Disable CC Without Being Logged Into YouTube Account???

### DISCUSSION THREAD

**Discussion by Blades07** [YouTube-Enhancer/extension#371] [Ideas]
Is it even possible to automatically disable CC on the YouTube player WITHOUT being logged into your YouTube account?  If so, this would be an awesome option to add to the YouTube Enhancer add-on.  Perhaps this could be done in the custom CSS settings, and if so, how???
I'm really sick and tired of going to YouTube to watch one video WITHOUT logging into my account and the CC being automatically turned on!  I'm just looking for a way to automatically disable CC without being logged into my account.  Thanks for any and all help with this!

**Comment by VampireChicken12** [comment]
@Blades07 i think that would be possible.

**Reply by Blades07** [reply]
❝𝒊 𝒕𝒉𝒊𝒏𝒌 𝒕𝒉𝒂𝒕 𝒘𝒐𝒖𝒍𝒅 𝒃𝒆 𝒑𝒐𝒔𝒔𝒊𝒃𝒍𝒆.❞

I've seen scrips that are supposed to enable CC, so I'm guessing that it can be disabled as well.  I took a look at these scripts in a text editor looking for a way to change a number value or whatever to make the script do the opposite of what it was created for, but I couldn't find any indication of what part of the script to edit.  Yes, I know, I should probably learn how to write scripts before I try to edit them.  🙄

**Comment by VampireChicken12** [comment]
@livingflore would you mind looking into implementing this feature.

**Reply by Blades07** [reply]
❝@𝒍𝒊𝒗𝒊𝒏𝒈𝒇𝒍𝒐𝒓𝒆 𝒘𝒐𝒖𝒍𝒅 𝒚𝒐𝒖 𝒎𝒊𝒏𝒅 𝒍𝒐𝒐𝒌𝒊𝒏𝒈 𝒊𝒏𝒕𝒐 𝒊𝒎𝒑𝒍𝒆𝒎𝒆𝒏𝒕𝒊𝒏𝒈 𝒕𝒉𝒊𝒔 𝒇𝒆𝒂𝒕𝒖𝒓𝒆.❞

Yes please, and thank you 𝐕𝐄𝐑𝐘 much!  👍👍

**Comment by Blades07** [comment]
I don't mean to be a pest, but I was wondering if the option to automatically disable the CC on YouTube is being looked into or being worked on?  
At this point, I'd seriously pay good money to have the ability to permanently disable this nonsense that YouTube either doesn't care about or is ramming it in people's face whether they want it or not!!!   >...>

**Reply by VampireChicken12** [reply]
I haven't looked into it but it seems possible. I have been busy.

**Reply by Blades07** [reply]
> I haven't looked into it but it seems possible. I have been busy.

Yeah, I would think that it would be, and that is why I'd LOVE to know how it could be done, specially since YouTube doesn't even honor this setting when you're logged into your YouTube account and have CC disabled.  
Oh, and I understand being busy, and I highly appreciate your replies.  Do you know of any good forums that I could go to and get help with this issue?  Thanks again for any and all help you may be able to provide!  👍

**Reply by VampireChicken12** [reply]
@Blades07 I'll look into implementing such a feature later today being Thursday September 26th

**Reply by VampireChicken12** [reply]
@Blades07 I have looked into how to enable/disable captions with JavaScript on the YouTube page. It is possible I will start working on the feature soon.
```js
// Get the player element
const playerContainer = isWatchPage() ? document.querySelector<YouTubePlayerDiv>("div#movie_player") : null;
// If player element is not available, return
if (!playerContainer) return;
// Disable captions
playerContainer.unloadModule("captions");
// Enable captions
playerContainer.loadModule("captions");
```

**Reply by VampireChicken12** [reply]
https://github.com/YouTube-Enhancer/extension/commit/00a4870f400b1897af1592b431708d7ed3ff7936
I released this feature in dev @Blades07 

## Release Artifacts
| File Name | SHA-256 Hash |
| :--- | :---: |
| [youtube-enhancer-v1.27.0-Chrome.zip](https://github.com/user-attachments/files/17155933/youtube-enhancer-v1.27.0-Chrome.zip) | 8d36a935fa81db0a721e94e08113355df54951cd74a5217edee383929d2a0e89 |
| [youtube-enhancer-v1.27.0-Edge.zip](https://github.com/user-attachments/files/17155936/youtube-enhancer-v1.27.0-Edge.zip) | 8d36a935fa81db0a721e94e08113355df54951cd74a5217edee383929d2a0e89 |
| [youtube-enhancer-v1.27.0-Firefox.zip](https://github.com/user-attachments/files/17155939/youtube-enhancer-v1.27.0-Firefox.zip) | c2d77efb747fcc301b239c55ef82bc89de674a1ad9efa94e3212628d9eea9d12 |

**Reply by VampireChicken12** [reply]
@Blades07 this feature has been released in version 1.27.0

**Comment by Blades07** [comment]
Thank you VERY much for this!  I'm confident that many other people like myself will highly appreciate this feature being added to the add-on.  Again, thank you and happy holidays.  👍👍

Cheers!

**Reply by VampireChicken12** [reply]
You're welcome, I feel the same. Happy holidays to you as well.
